### PR TITLE
Fixes of ProgressDialog is deprecated

### LINKED ISSUE #3336: ```ProgressDialog``` is deprecated
Parent Issue #3314

**Describe the bug**
 We are using `ProgressDialog` in `ZimHostFragment` while we are starting the kiwix hotspot. This method is now deprecated and the suggested replacement is to use `ProgressBar` or we can create a custom view for it and show it in the dialog.

**Logs**
'ProgressDialog' is deprecated. Deprecated in Java

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
Fixes #3336 

We have changed the deprecated `ProgressDailog` with `KiwixDialog` and made a custom layout to show the progress to the user.

https://github.com/kiwix/kiwix-android/assets/34593983/b55b27f9-4484-4000-b070-ba47e1551bd3

https://github.com/kiwix/kiwix-android/assets/34593983/0986c146-dd0d-432d-8a2f-008a6c039d5b

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/utils/dialog/KiwixDialog.kt [hunk 1]
```diff
@@ -22,10 +22,11 @@ import android.app.Activity
 import android.view.View
 import org.kiwix.kiwixmobile.core.R
 
+@Suppress("LongParameterList")
 sealed class KiwixDialog(
   val title: Int?,
   val message: Int?,
-  val positiveMessage: Int,
+  val positiveMessage: Int?,
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/utils/dialog/KiwixDialog.kt:29]: Having a not-optional field is a good idea to make sure the user knows what's going on. It feels weird if we don't have any message to say users what's going on. What do you think?

File: core/src/main/java/org/kiwix/kiwixmobile/core/utils/dialog/KiwixDialog.kt [hunk 2]
```diff
@@ -227,6 +228,14 @@ sealed class KiwixDialog(
     getView = customGetView
   )
 
+  data class StartServer(val customGetView: (() -> View)?) : KiwixDialog(
+    R.string.progress_dialog_starting_server,
+    null,
+    null,
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/utils/dialog/KiwixDialog.kt:234]: If you are creating a null type to pass the null. We can pass an empty string here.

**Review by gouri-panda — COMMENTED** [review]
@MohitMaliFtechiz Just a small number of requests then we are good to go.

**Review by gouri-panda — COMMENTED** [review]
@MohitMaliFtechiz Just a small number of requests then we are good to go.

File: core/src/main/java/org/kiwix/kiwixmobile/core/utils/dialog/KiwixDialog.kt (hunk 1 above)
Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/utils/dialog/KiwixDialog.kt:29]: Good point :+1:

File: core/src/main/java/org/kiwix/kiwixmobile/core/utils/dialog/KiwixDialog.kt (hunk 2 above)
Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/utils/dialog/KiwixDialog.kt:234]: @gouri-panda, We can not directly pass an empty string here as our positive message parameter takes an Int `val positiveMessage: Int` and we need to pass a valid string resource value to it. But we can make an empty string resource and pass it here.

Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/utils/dialog/KiwixDialog.kt]: Done.

**Comment by kelson42** [comment]
@MohitMaliFtechiz Like I said, if this is a spinner, call it a spinner. If this is a progressbar, call it a progressbar. AFAIK, we have a spinner, called a progressbar. Please fix.

**Comment by kelson42** [comment]
@gouri-panda Looks good to you?

**Pull request merged**